rust/lance/src/dataset.rs

```diff
@@ -528,9 +533,16 @@ impl Dataset {
     ///
     pub async fn write(
         batches: impl RecordBatchReader + Send + 'static,
-        dest: impl Into<WriteDestination<'_>>,
+        dest: impl Into<WriteDestination<'_>> + Clone,
         params: Option<WriteParams>,
     ) -> Result<Self> {
+        if let Some(dataset) = dest.clone().into().dataset() {
+            info!(target: TRACE_DATASET_EVENTS, event=DATASET_WRITING_EVENT, uri=dataset.uri());
```

Comment:
I remember we discussed also recording the insert mode here, looks like that was lost in the latest revision